verification/cbmc/proofs/aws_cryptosdk_private_algorithm_message_id_len/aws_cryptosdk_private_algorithm_message_id_len_harness.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/cipher.h>
+#include <make_common_data_structures.h>
+
+void aws_cryptosdk_private_algorithm_message_id_len_harness() {
+    /* Nondet Input */
+    const struct aws_cryptosdk_alg_properties *props = malloc(sizeof(*props));
```

Comment:
We need to use a proper allocator for this structure.
```c
struct aws_cryptosdk_alg_properties {
    const char *md_name;
    const char *cipher_name;
    const char *alg_name;
    const struct aws_cryptosdk_alg_impl *impl;
    size_t data_key_len;
    size_t content_key_len;
    size_t iv_len;
    size_t tag_len;
    size_t signature_len;
    enum aws_cryptosdk_alg_id alg_id;
    enum aws_cryptosdk_hdr_version msg_format_version;
    size_t alg_suite_data_len;
    size_t commitment_len;
    const char *sig_md_name;
};
```
Every pointer must be properly allocated as well.

Reply:
I see that you have a `ensure_alg_properties_attempt_allocation` function. You might add this allocation also inside the function.

Reply:
We should allocate properly all pointers in `impl`, yes.